src/ng/httpBackend.js

```diff
@@ -120,11 +120,15 @@ function createHttpBackend($browser, createXhr, $browserDefer, callbacks, rawDoc
       };
 
       xhr.onerror = requestError;
-      xhr.onabort = requestAborted;
       xhr.ontimeout = requestTimeout;
 
+      // The timeout behavior follows standard XMLHttpRequest logic.
+      // xhr.timeout = ontimeout (numerical timeout or $timeout)
+      // xhr.abort() = onabort (resolving a promise)
+      xhr.onabort = timeout > 0 || (timeout && timeout.$$timeoutId >= 0) ? requestTimeout : requestAborted;
```

Comment:
AFAICT, this will call the wrong method if the request is aborted while there is a timeout set (even if the "abortion" was not related to the timeout), right?

I think we should set a flag in `requestTimeout()` and adjust the `xhrStatus` argument in `requestAborted()` based on the value of that flag.

Reply:
Not sure if I understand you correctly, but as it goes, at the moment you can only have a numerical timeout and the ability to cancel the request before that if you pass a promise into `timeout` and wire up your own timeout behind that promise (Because you cannot short-circuit a $timeout). *And*  in that case, you cannot get the correct xhrStatus, because the xhrStatus is based on the value that is passed into the `timeout`, that is correct. 
Based on the resolve value of a promise-like, we could set the correct xhrStatus, that is right.

Reply:
What I mean is that if something other than the timeout (numeric or promise) causes the request to be aborted (e.g. someone calls `xhr.abort()` manually or the browser aborts it - not sure if browsers do that), then your current implementation will call `requestTimeout()`, even if the `abort` event was not related to a timeout.

Reply:
Oh, I see.
As we don't expose the xhr object (or do we via $xhrFactory?), I don't think it can happen that a user aborts the request.
I think when the browser unloads a page (because of navigation), then onabort gets called, but in most cases it's too late for the code to react.

But I guess it's safer if the default is abort for onabort.

Reply:
Yeah, with `$xhrFactory()` the user can get access to the `xhr` object :grin: